torchmetrics/timeseries.py

```diff
@@ -0,0 +1,48 @@
+from typing import List, Tuple, Union
+
+import torch
+from torch import nn
+
+from torchmetrics.metric import Metric
+
+
+class TimeSeriesMetric(nn.ModuleList):
+    """
+    A class that keeps track of metrics as a timeseries, and implements useful methods.
+
+    Args:
+        metric: the torchmetric to keep track of at each timestep.
+    """
+
+    def __init__(self, metric: Metric):
+        super().__init__()
+        self.metric = metric
+
+    @property
+    def n_timesteps(self) -> int:
+        return len(self)
+
+    def add_timestep(self, preds: torch.Tensor, targets: torch.Tensor) -> None:
```

Comment:
to make sure that this wrapper works with all metrics I think we need to keep it very general
```suggestion
    def add_timestep(self, *args, **kwargs) -> None:
```